lib/runner.js

```diff
@@ -47,6 +47,15 @@ function Runner(opts) {
 	EventEmitter.call(this);
 
 	this.options = opts || {};
+
+	if (this.options.grep instanceof RegExp) {
+		this.grep = this.options.grep;
```

Comment:
I don't think the API should support both regex and string. It should be converted from a string to regex in the CLI.